pkg/apis/controller/v1beta1/schedulingspec.go

```diff
@@ -45,17 +45,46 @@ type SchedulingSpec struct {
 	Spec SchedulingSpecTemplate `json:"spec,omitempty" protobuf:"bytes,1,rep,name=spec"`
 }
 
+type ClusterReference struct {
+	Name string `json:"name"`
+}
+
+type ClusterSchedulingSpec struct {
+	Clusters        []ClusterReference    `json:"clusters,omitempty"`
+	ClusterSelector *metav1.LabelSelector `json:"clusterSelector,omitempty"`
+}
+
+type ScheduleTimeSpec struct {
+	Min     metav1.Time `json:"minTimestamp,omitempty"`
+	Desired metav1.Time `json:"desiredTimestamp,omitempty"`
+	Max     metav1.Time `json:"maxTimestamp,omitempty"`
+}
+
+type DispatchDurationSpec struct {
+	Expected int `json:"expected,omitempty"`
```

Comment:
I don't see expected used anywhere in the `queuejob_controller_ex.go` so should remove

Reply:
This will be used in future